pkg/provisioner/ironic/servicing.go

```diff
@@ -0,0 +1,124 @@
+package ironic
+
+import (
+	"fmt"
+
+	"github.com/gophercloud/gophercloud/v2/openstack/baremetal/v1/nodes"
+	"github.com/metal3-io/baremetal-operator/pkg/hardwareutils/bmc"
+	"github.com/metal3-io/baremetal-operator/pkg/provisioner"
+)
+
+func (p *ironicProvisioner) buildServiceSteps(bmcAccess bmc.AccessDetails, data provisioner.ServicingData) (serviceSteps []nodes.ServiceStep, err error) {
+	// Get the subset (currently 3) of vendor specific BIOS settings converted from common names
+	var firmwareConfig *bmc.FirmwareConfig
+	if data.FirmwareConfig != nil {
+		bmcConfig := bmc.FirmwareConfig(*data.FirmwareConfig)
+		firmwareConfig = &bmcConfig
+	}
+	fwConfigSettings, err := bmcAccess.BuildBIOSSettings(firmwareConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	newSettings := p.getNewFirmwareSettings(data.ActualFirmwareSettings, data.TargetFirmwareSettings, fwConfigSettings)
+	if len(newSettings) != 0 {
+		p.log.Info("Applying BIOS config clean steps", "settings", newSettings)
+		serviceSteps = append(
+			serviceSteps,
+			nodes.ServiceStep{
+				Interface: nodes.InterfaceBIOS,
+				Step:      "apply_configuration",
+				Args: map[string]interface{}{
+					"settings": newSettings,
+				},
+			},
+		)
+	}
+
+	// TODO: Add service steps for firmware updates
```

Comment:
We don't have the logic for the firmware updates  here, so we won't do any action at all.
You can get some inspiration on  https://github.com/metal3-io/baremetal-operator/blob/main/pkg/provisioner/ironic/ironic.go#L1250-L1277